.github/workflows/base.yml

```diff
@@ -344,18 +363,47 @@ jobs:
           cd quarkus
           bash ../workflow-mandrel/.github/update_quarkus_version.sh 2.2.999
         fi
-    # Use Java 17 to build Quarkus as it doesn't build with Java 20
+    # Use Java 11 to build Quarkus as that's the lowest supported JDK version currently
     - uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
-        java-version: '17'
+        java-version: '11'
+    - name: Download GraalVM Maven Repo
+      if: "!startsWith(matrix.os-name, 'windows')"
+      uses: actions/download-artifact@v3
+      with:
+        name: org-graalvm-artefacts-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
+        path: .
+    - name: Download GraalVM Maven Version
+      if: "!startsWith(matrix.os-name, 'windows')"
+      uses: actions/download-artifact@v3
+      with:
+        name: mandrel-maven-version-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
+        path: .
+    - name: Extract GraalVM Maven Repo and GraalVM Maven Version
+      if: "!startsWith(matrix.os-name, 'windows')"
+      shell: bash
+      run: |
+        tar -xzvf graalvm-maven-artefacts.tgz -C ~
+        tar -xzvf graalvm-version.tgz -C $(dirname ${MANDREL_HOME})
     - name: Build quarkus
       run: |
+        rm -f maven_graalvm_before_build.txt maven_graalvm_after_build.txt
+        find ~/.m2/repository/org/graalvm | sort > maven_graalvm_before_build.txt
+        GRAAL_MVN_ARTIFACTS_VERS=$(cat ${MANDREL_HOME}/.maven-version)
+        ls ${MANDREL_HOME}/.maven-version
+        echo "Building quarkus with locally installed GraalVM maven artefacts in version: ${GRAAL_MVN_ARTIFACTS_VERS}"
         cd ${QUARKUS_PATH}
-        ./mvnw -e -B --settings ${QUARKUS_PATH}/.github/mvn-settings.xml  -Dquickly
+        ./mvnw -e -B --settings ${QUARKUS_PATH}/.github/mvn-settings.xml -Dquickly -Dgraal-sdk.version="${GRAAL_MVN_ARTIFACTS_VERS}"
+        cd -
+        find ~/.m2/repository/org/graalvm | sort > maven_graalvm_after_build.txt
+        diff -u maven_graalvm_before_build.txt maven_graalvm_after_build.txt
     - name: Tar Maven Repo
       shell: bash
-      run: tar -czvf maven-repo.tgz -C ~ .m2/repository
+      run: |
+        # Don't include locally built graal maven artefacts for native IT runs
```

Comment:
Why not? Is it in order to keep things simple?

Reply:
Yes, this is beyond the scope of this feature. The idea is to recognize build failures due to broken bytecode versions in graalvm maven artefacts early.

Native IT tests should just continue to download the version referenced in the BOM of the relevant Quarkus version.

Reply:
Well, I stand corrected :-) We have to include it as the quarkus BOM will have that version referenced. See the CI failure here: https://github.com/jerboaa/graal/actions/runs/6146546436/job/16677606253#step:12:71

Reply:
Updated PR no longer removes it, which should fix that problem.

Reply:
Oh, indeed I didn't think about this.